Migrate to manifest v3

Aarav-Juneja opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [x] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
-->

### Summary
Updated extension to manifest v3. This should resolve #52.


### Details and comments
I didn't completely understand the code, so I may have done something wrong.
I deleted the `background.js` file in the extension. I found this code to be duplicated in the `popup.js` file. I also removed the `content.js` from the background scripts because it was already labeled as a content script.
Can I also get this PR under hacktoberfest.
Lastly, I am unsure what, if any, documentation changes are needed.

gobbledy-gook commented:
I would like to see you try to unpack this extension on your own system and check if this is working perfectly fine. Share a screenshot of the same.

Aarav-Juneja commented:
I found a bug and have now fixed it. We were reloading the popup file every page to ensure the data is up to date. I have now changed it so that the popup 'asks' the content script on the page for the data. This increases performance, because the extension only reloads when necessary.
What photos would you like? - There are no errors in console.

gobbledy-gook commented:
@Aarav-Juneja good update guys ! I have reviewed it and works fine. 

For some reason it does show some error for anonymous asyn function that has been used in ```popup.js```. Here is a screenshot of this error. @AnsahMohammad any inputs on this error ?

![Screenshot 2023-10-21 at 10 49 13 PM](https://github.com/gobbledy-gook/report.ai/assets/96362727/0334e81d-5bd5-4e5b-91cb-7aca2bab1b51)
